src/map/map.ts

```diff
@@ -572,40 +572,30 @@ export class PropertiesMap {
     private _createPlot() {
         this._plot.innerHTML = '';
 
-        const colors = this._colors();
-        const lineColors = this._lineColors();
-        const sizes = this._sizes();
-        const symbols = this._symbols();
-
-        const x = this._coordinates(this._options.x);
-        const y = this._coordinates(this._options.y);
-        const z = this._coordinates(this._options.z);
-
         const type = this._is3D() ? 'scatter3d' : 'scattergl';
-
         // The main trace, containing default data
         const main = {
             name: '',
             type: type,
 
-            x: x[0],
-            y: y[0],
-            z: z[0],
+            x: this._coordinates(this._options.x, 0)[0],
+            y: this._coordinates(this._options.y, 0)[0],
+            z: this._coordinates(this._options.z, 0)[0],
 
             hovertemplate: this._options.hovertemplate(),
             marker: {
-                color: colors[0],
+                color: this._colors(0)[0],
                 coloraxis: 'coloraxis',
                 line: {
-                    color: lineColors[0],
+                    color: this._lineColors(0)[0],
                     width: 1,
                 },
                 // prevent plolty from messing with opacity when doing bubble
                 // style charts (different sizes for each point)
                 opacity: 1,
-                size: sizes[0],
+                size: this._sizes(0)[0],
                 sizemode: 'area',
-                symbol: symbols[0],
+                symbol: this._symbols(0)[0],
```

Comment:
@rosecers The core of the issue with `symbolFunc is not a function` was that the interface of Plotly.createPlot & Plotly.restyle are different. The first one takes a list of traces, containing `x: [1, 2, 3, 4]`; while the second one takes a list of elements to change, containing one array for each trace `x: [[1, 2, 3, 4], [...], [...]]`. Trying to use `this._symbols(0)` in `_createPlot` results in `symbols: ['circle']`, but we want `symbols: 'circle'`.

Let me know if I'm unclear